From 0bd71faf584365a9507f441f02c54eec9c3a7110 Mon Sep 17 00:00:00 2001
From: Dylan Hung <dylan_hung@aspeedtech.com>
Date: Fri, 30 Sep 2022 15:21:30 +0800
Subject: [PATCH 24/30] i3c: aspeed: reject hot-join by default

I3C master code does not support hot-join yet.  So just NACK the
hot-join request by default.

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I6642170ad79b2bdc47ab565e8296314b0fdee90a
---
 drivers/i3c/i3c_aspeed.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i3c/i3c_aspeed.c b/drivers/i3c/i3c_aspeed.c
index dbba405792..2676443032 100644
--- a/drivers/i3c/i3c_aspeed.c
+++ b/drivers/i3c/i3c_aspeed.c
@@ -1077,6 +1077,7 @@ static void i3c_aspeed_enable(struct i3c_aspeed_obj *obj)
 
 	reg.value = i3c_register->device_ctrl.value;
 	reg.fields.enable = 1;
+	reg.fields.hj_ack_ctrl = 1;
 	reg.fields.slave_ibi_payload_en = 1;
 	if (config->secondary) {
 		reg.fields.slave_auto_mode_adapt = 0;
-- 
2.24.1

